src/Utils/UseGetAxios.jsx

```diff
@@ -0,0 +1,30 @@
+export const UseGetAxios = (api, encodedToken) => {
+    const [isLoading, setIsLoading] = useState(true);
+    const [serverData, setServerData] = useState({});
+    const [serverError, setServerError] = useState(null);
+    const getData = async () => {
+        try {
+            setIsLoading(true);
+            if (encodedToken) {
+                let { data } = await axios.get(api, { headers: { authorization: encodedToken } });
+            }
+            else {
+                let { data } = await axios.get(api);
```

Comment:
instead of destructuring the data, you should return the whole response, because you might need to check the response status sometimes. If you destructure only the data, then you won't get be able to access the response status and other info that comes from the server

Reply:
we are already receiving it in serverError if it fails to get, we can check from it